### PR TITLE
S03 us12 connexion page

### PR DESCRIPTION
**Pull request by pjroggy**

### CODE REVIEW

File: client/src/main.jsx [hunk 1]
```diff
@@ -38,6 +42,62 @@ const router = createBrowserRouter([
       {
         path: "/connexion",
         element: <LoginPage />,
+        action: async ({ request }) => {
+          try {
+            const formData = await request.formData();
+            const data = Object.fromEntries(formData.entries());
+            console.info(formData);
```
Comment (FGAT3) [inline client/src/main.jsx]: no

File: client/src/main.jsx [hunk 2]
```diff
@@ -38,6 +42,62 @@ const router = createBrowserRouter([
...
+
+            const response = await sendData(url, data, "POST");
+
+            if (response && response.ok) {
+              return redirect(`/`);
+            }
+
+            throw new Error("Invalid response from server");
+          } catch (error) {
+            console.error("Error submitting form:", error);
```
Comment (FGAT3) [inline client/src/main.jsx]: no

File: client/src/main.jsx [hunk 3]
```diff
@@ -38,6 +42,62 @@ const router = createBrowserRouter([
...
+        },
+      },
+      {
+        path: "/inscription",
+        element: <RegistrationPage />,
+        action: async ({ request }) => {
+          try {
+            const formData = await request.formData();
+            const data = Object.fromEntries(formData.entries());
+            console.info(data)
```
Comment (FGAT3) [inline client/src/main.jsx]: no

File: client/src/main.jsx [hunk 4]
```diff
@@ -38,6 +42,62 @@ const router = createBrowserRouter([
...
+
+            const response = await sendData(url, data, "post");
+
+            if (response && response.ok) {
+              return redirect(`/connexion`);
+            }
+
+            throw new Error("Invalid response from server");
+          } catch (error) {
+            console.error("Error submitting form:", error);
```
Comment (FGAT3) [inline client/src/main.jsx]: no

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    email: "",
+    password: "",
+  });
+
+  const handleLoginChange = (event) => {
+    setFormLogin({ ...formLogin, [event.target.name]: event.target.value });
+  };
+
+  const postLogin = () => {
+    console.info(formLogin);
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.jsx]: no

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        <label className="label-connexion" htmlFor="email">
+          Email
+        </label>
+        <input
+          className="input-connexion"
+          type="email"
+          id="email"
+          name="email"
+          value={formLogin.email}
+          placeholder="johndoe@email.com"
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.jsx]: change it for something like "Entrez votre email"

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+          pattern="^[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}$"
+          required
+          onChange={handleLoginChange}
+        />
+        <label className="label-connexion" htmlFor="password">
+          Mot de passe
+        </label>
+        <input
+          className="input-connexion"
+          type="text"
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.jsx]: input type password

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+          value={formLogin.password}
+          placeholder="********"
+          required
+          onChange={handleLoginChange}
+        />
+        <button
+          className="login-button"
+          type="submit"
+          aria-label="Se connecter"
+          onClick={postLogin}
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.jsx]: not necessary

File: client/src/pages/LoginPage/LoginPage.scss [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  text-align: center;
+}
+.connexion-form {
+  display: flex;
+  flex-direction: column;
+  margin: 1.2rem;
+  border-radius: 0.4rem;
+}
+.label-connexion {
+  font-weight: 700;
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.scss]: too much better use 600 I think

File: client/src/pages/LoginPage/LoginPage.scss [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  color: $black-color;
+  background-color: $white-color;
+  margin: 1rem auto;
+  padding-top: 0.15rem;
+  border: 0.2rem outset $secondary-color;
+  border-radius: 0.7rem;
+  width: 10rem;
+  height: 1.6rem;
+  font-size: 0.8rem;
+  font-weight: 450;
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.scss]: 500 or 400 plz

File: client/src/pages/LoginPage/LoginPage.scss [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+}
+.link-registration:hover {
+  background-color: rgb(221, 221, 221);
```
Comment (FGAT3) [inline client/src/pages/LoginPage/LoginPage.scss]: no

Comment (FGAT3) [reply client/src/pages/LoginPage/LoginPage.scss]: hexadécimal plz

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+function RegistrationPage() {
+  const [formRegistration, setFormRegistration] = useState({
+    firstname: "",
+    lastname: "",
+    pseudo: "",
+    email: "",
+    password: "",
+  });
+
+  const handleRegistrationChange = (event) => {
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: Is it usefull ?

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+    <>
+      <h2 className="registration-title">Inscription </h2>
+      <Form method="post" className="registration-form">
+        <label className="label-registration" htmlFor="firstname">
+          Prénom
+        </label>
+        <input
+          className="input-registration"
+          type="text"
+          minLength="5"
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: not necessary

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+          required
+          onChange={handleRegistrationChange}
+        />
+        <label className="label-registration" htmlFor="lastname">
+          Nom
+        </label>
+        <input
+          className="input-registration"
+          type="text"
+          minLength="5"
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: not necessary

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+          required
+          onChange={handleRegistrationChange}
+        />
+        <label className="label-registration" htmlFor="pseudo">
+          Pseudo
+        </label>
+        <input
+          className="input-registration"
+          type="text"
+          minLength="5"
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: no

File: client/src/pages/RegistrationPage/RegistrationPage.scss [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+  border-radius: 0.4rem;
+  background-image: linear-gradient(#12022b94, #12022b94),
+    url("https://assets.codepen.io/13471/sparkles.gif");
+
+  @include for-desktop-up {
+    margin: auto 5rem;
+  }
+}
+.label-registration {
+  font-weight: 700;
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.scss]: would be harmonized with the rest of the code to use 600

File: client/src/pages/RegistrationPage/RegistrationPage.scss [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+  margin: 0.5rem;
+  @include for-desktop-up {
+    margin: 0.5rem 10rem;
+  }
+}
+.input-registration {
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.scss:12]: applicate this style one multiple inputs or create mixin for not dupplicate css

File: client/src/pages/RegistrationPage/RegistrationPage.scss [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  padding-top: 0.15rem;
+  border: 0.2rem outset $secondary-color;
+  border-radius: 0.7rem;
+  width: 10rem;
+  height: 1.6rem;
+  font-size: 0.8rem;
+  font-weight: 450;
+}
+.link-connexion:hover {
+  background-color: rgb(221, 221, 221);
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.scss]: use hexadecimal plz

File: client/src/pages/RegistrationPage/RegistrationPage.scss [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+  color: $black-color;
+  background-color: $white-color;
+  margin: 1rem auto;
+  padding-top: 0.15rem;
+  border: 0.2rem outset $secondary-color;
+  border-radius: 0.7rem;
+  width: 10rem;
+  height: 1.6rem;
+  font-size: 0.8rem;
+  font-weight: 450;
```
Comment (FGAT3) [inline client/src/pages/RegistrationPage/RegistrationPage.scss]: use 400 or 500

File: server/app/controllers/userAction.js [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+const read = async (req, res, next) => {
+  try {
+    const user = await tables.user.read(req.params.id);
+
+    if (user == null) {
+      res.sendStatus(404);
+    } else {
+      res.status(200).json(user);
```
Comment (FGAT3) [inline server/app/controllers/userAction.js:20]: you could use this status in catch error inside main.jsx

File: server/app/middelware/verifyToken.js [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+      throw new Error("Authorization header has not the 'Bearer' type");
+    }
+
+    // Vérifier la validité du token (son authenticité et sa date d'expériation)
+    // En cas de succès, le payload est extrait et décodé
+    req.auth = jwt.verify(token, process.env.APP_SECRET);
+
+    next();
+  } catch (err) {
+    console.error(err);
```
Comment (FGAT3) [inline server/app/middelware/verifyToken.js:25]: no

File: server/database/models/UserRepository.js [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+
+  async create(user) {
+    console.info(user);
+    // Execute the SQL INSERT query to add a new user to the "user" table
+    const [result] = await this.database.query(
+      `INSERT INTO ${this.table} (
+        created_at,
+        firstname,
+        lastname,
+        avatar_image,
```
Comment (FGAT3) [inline server/database/models/UserRepository.js:21]: a user doesn't select an avatar when creating account so not necessary

File: server/database/models/UserRepository.js [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  // The C of CRUD - Create operation
+
+  async create(user) {
+    console.info(user);
+    // Execute the SQL INSERT query to add a new user to the "user" table
+    const [result] = await this.database.query(
+      `INSERT INTO ${this.table} (
+        created_at,
```
Comment (FGAT3) [inline server/database/models/UserRepository.js:18]: created automatically I think

**Review by FGAT3 — COMMENTED** [review]
I count on your good work

File: client/src/pages/LoginPage/LoginPage.scss (hunk 10 above)
Comment (pjroggy) [reply client/src/pages/LoginPage/LoginPage.scss]: 500

File: server/database/models/UserRepository.js [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+        role = ?`,
+      [
+        user.created_at,
+        user.firstname,
+        user.lastname,
+        user.avatar_image,
+        user.pseudo,
+        user.email,
+        user.hashedPassword,
+        user.role,
```
Comment (FGAT3) [inline server/database/models/UserRepository.js]: User will not be able to modify his role

**Review by FGAT3 — APPROVED** [review]

**Review by JadeJolivet — APPROVED** [review]
awesome !

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+          className="input-registration"
+          type={showPassword ? "text" : "password"}
+          minLength="5"
+          id="password"
+          name="password"
+          placeholder="Mot de passe"
+          required
+        />
+        <label className="label-registration" htmlFor="showPassword">
+          <img src="" alt="" />
```
Comment (KevinDavoust) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: à finir

**Pull request merged**